### PR TITLE
Remove usages of beta Guava APIs.

### PR DESCRIPTION
**Pull request by anuraaga**
Long term we have #1944 but in the meantime, we definitely can't be using beta APIs so removed them and added a checker.

### CODE REVIEW

File: sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 1]
```diff
@@ -373,7 +372,9 @@ private void addTimedEvent(Event timedEvent) {
         logger.log(Level.FINE, "Calling addEvent() on an ended Span.");
         return;
       }
-      events.add(timedEvent);
+      if (events.size() < traceConfig.getMaxNumberOfEvents()) {
```
Comment (anuraaga) [inline sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:375]: Spec just says discard after reaching the limit, which is closer to what this code does than the previous FIFO queue (though personally I think either is fine)

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#limits-on-span-collections

File: sdk/common/src/main/java/io/opentelemetry/sdk/common/DaemonThreadFactory.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public DaemonThreadFactory(String namePrefix) {
 
   @Override
   public Thread newThread(Runnable runnable) {
-    Thread t = MoreExecutors.platformThreadFactory().newThread(runnable);
+    Thread t = Executors.defaultThreadFactory().newThread(runnable);
```
Comment (anuraaga) [inline sdk/common/src/main/java/io/opentelemetry/sdk/common/DaemonThreadFactory.java:26]: Previous code has some checks for Google App Engine - no clue what the delta is but this seems reasonable enough.

Comment (jkwatson) [reply sdk/common/src/main/java/io/opentelemetry/sdk/common/DaemonThreadFactory.java:26]: I peeked at this a few weeks ago. It looks like it's old app-engine stuff, so I'm not too worried about it.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**